tests/ui/parser/issues/issue-22647.rs

```diff
@@ -1,5 +1,5 @@
 fn main() {
-    let caller<F> = |f: F|  //~ ERROR expected one of `:`, `;`, `=`, `@`, or `|`, found `<`
+    let caller<F> = |f: F|  //~ ERROR generic args in patterns require the turbofish syntax
```

Comment:
Looking at this error, the suggestion we should be giving is to add a type, but don't know how hard it would be to change.

Reply:
Like function parameters, maybe we can also suggest `name: caller<F>` or `_: caller<F>`? I think it may be more general, because `caller<F>` is more like a type.

Reply:
Yeah, I think the later is the better suggestion (particularly if the rest can be parsed as a type until `;` or `=` is reached).